reference-adapters/mukta-ifix-adapter/src/main/java/org/egov/works/mukta/adapter/kafka/MuktaAdaptorConsumer.java

```diff
@@ -67,24 +65,24 @@ public void listen(final String record, @Header(KafkaHeaders.RECEIVED_TOPIC) Str
         try {
             log.info("Payment data received on.");
             paymentRequest = objectMapper.readValue(record, PaymentRequest.class);
-            log.info("Payment data is " + paymentRequest);
-            isRevised = disbursementValidator.isValidForDisbursementCreate(paymentRequest);
-            Disbursement disbursement = paymentInstructionService.processPaymentInstruction(paymentRequest,isRevised);
-            encryptedDisbursement = paymentInstructionEnrichment.encriptDisbursement(disbursement);
-            pi = paymentInstructionEnrichment.getPaymentInstructionFromDisbursement(encryptedDisbursement);
-            msgHeader = programServiceUtil.getMessageCallbackHeader(paymentRequest.getRequestInfo(), muktaAdaptorConfig.getStateLevelTenantId());
-            msgHeader.setAction(Action.CREATE);
-            msgHeader.setMessageType(MessageType.DISBURSE);
-            DisbursementRequest disbursementRequest = DisbursementRequest.builder().header(msgHeader).message(disbursement).build();
-            DisbursementRequest encriptedDisbursementRequest = DisbursementRequest.builder().header(msgHeader).message(encryptedDisbursement).build();
-            muktaAdaptorProducer.push(muktaAdaptorConfig.getDisburseCreateTopic(), encriptedDisbursementRequest);
-            paymentInstructionService.updatePIIndex(paymentRequest.getRequestInfo(), pi,isRevised);
-            try {
-                Thread.sleep(5000);
-            } catch (InterruptedException e) {
-                throw new RuntimeException(e);
+            Boolean isDisbursementCreated = disbursementValidator.isDisbursementCreated(paymentRequest);
+            if(Boolean.FALSE.equals(isDisbursementCreated))
+            {
+                log.info("Payment data is " + paymentRequest);
+                isRevised = disbursementValidator.isValidForDisbursementCreate(paymentRequest);

```

Comment:
instead of instantiating all these variables outside, can't we do it within the try block itself? why are we instantiating them outside?